torchvision/models/maxvit.py

```diff
@@ -0,0 +1,595 @@
+        expansion_ratio: float,
+        # conv + transformer parameters
+        # normalization_fn is applied only to the conv layers
+        # activation_fn is applied both to conv and transformer layers
+        normalization_fn: Callable[..., nn.Module],
+        activation_fn: Callable[..., nn.Module],
+        # transformer parameters
+        head_dim: int,
+        mlp_ratio: int,
+        mlp_dropout: float,
+        attn_dropout: float,
+        # partitioning parameters
+        partition_size: int,
+    ) -> None:
+        super().__init__()
+
+        # stem
+        self.stem = nn.Sequential(
+            Conv2dNormActivation(
+                input_channels, stem_channels, 3, stride=2, norm_layer=None, activation_layer=None, bias=True
+            ),
+            Conv2dNormActivation(
+                stem_channels, stem_channels, 3, stride=1, norm_layer=None, activation_layer=None, bias=True
+            ),
+        )
+
+        # account for stem stride
+        input_size = (input_size[0] // 2, input_size[1] // 2)
+
+        # blocks
+        self.blocks = nn.ModuleList()
+        in_channels = [stem_channels] + block_channels[:-1]
+        out_channels = block_channels
+
+        # precompute the stochastich depth probabilities from 0 to stochastic_depth_prob
+        # since we have N blocks with L layers, we will have N * L probabilities uniformly distributed
+        # over the range [0, stochastic_depth_prob]
+        p_stochastic = np.linspace(0, stochastic_depth_prob, num=sum(block_layers)).tolist()
+
+        p_idx = 0
+        for in_channel, out_channel, num_layers in zip(in_channels, out_channels, block_layers):
+            self.blocks.append(
+                MaxVitBlock(
+                    in_channels=in_channel,
+                    out_channels=out_channel,
+                    squeeze_ratio=squeeze_ratio,
+                    expansion_ratio=expansion_ratio,
+                    normalization_fn=normalization_fn,
+                    activation_fn=activation_fn,
+                    head_dim=head_dim,
+                    mlp_ratio=mlp_ratio,
+                    mlp_dropout=mlp_dropout,
+                    attn_dropout=attn_dropout,
+                    partition_size=partition_size,
+                    input_grid_size=input_size,
+                    n_layers=num_layers,
+                    p_stochastic=p_stochastic[p_idx : p_idx + num_layers],
+                ),
+            )
+            input_size = self.blocks[-1].grid_size
+            p_idx += num_layers
+
+        self.classifier = nn.Sequential(
+            nn.AdaptiveAvgPool2d(1),
+            nn.Flatten(),
+            nn.Linear(block_channels[-1], out_classes, bias=False),
+        )
+
+    def forward(self, x: Tensor) -> Tensor:
+        x = self.stem(x)
+        for block in self.blocks:
+            x = block(x)
+        x = self.classifier(x)
+        return x
+
+
+def max_vit_T_224(num_classes: int) -> MaxVit:
```

Comment:
Add an Enum for the weights. You can have them empty like:

```python
class MaxVit_T_Weights(WeightsEnum):
    pass
```

Also I recommend renaming the `max_vit_T_224` to `maxvit_t` because:
- It will follow the standards of other classes
- we shouldn't hard code the input size in the method. (I'm aware the the model operates on fixed input size. See [ViT](https://github.com/pytorch/vision/blob/main/torchvision/models/vision_transformer.py) for similar approach)

Finally, make sure you register the model builder using the new API, so that it will appear on your tests. Here is how it's done: #6330. Feedback is very welcome concerning the mechanism as it's very new.